runtime/str.go

```diff
@@ -711,8 +755,6 @@ func adjustIndex(i, l int) int {
 		i = 0
 	case i < 0:
 		i += l
-	case i > l:
-		i = l
```

Comment:
Maybe the adjustIndex helper function isn't very useful and we should just do the logic inline? I'm not really sure.